packages/storage/src/indexed-db/index.ts

```diff
@@ -264,4 +271,63 @@ export class IndexedDb implements IndexedDbInterface {
   async saveGasPrices(value: GasPrices): Promise<void> {
     await this.u.update({ table: 'GAS_PRICES', value, key: 'gas_prices' });
   }
+
+  async getNotesForVoting(
+    addressIndex: AddressIndex | undefined,
+    votable_at_height: bigint,
+  ): Promise<NotesForVotingResponse[]> {
+    const relevantAssets = new Map<string, DenomMetadata>();
+
+    let assetCursor = await this.db.transaction('ASSETS').store.openCursor();
+
+    while (assetCursor) {
+      const isDelegationAssetType = new RegExp('udelegation_.*');
+
+      const denomMetadata = DenomMetadata.fromJson(assetCursor.value);
+
+      if (isDelegationAssetType.test(denomMetadata.base) && denomMetadata.penumbraAssetId) {
+        relevantAssets.set(uint8ArrayToHex(denomMetadata.penumbraAssetId.inner), denomMetadata);
+      }
+      assetCursor = await assetCursor.continue();
+    }
+
+    const notesForVoting = [];
+
+    let noteCursor = await this.db.transaction('SPENDABLE_NOTES').store.openCursor();
+    while (noteCursor) {
+      const note = SpendableNoteRecord.fromJson(noteCursor.value);
+
+      if (addressIndex) {
+        if (!note.addressIndex?.equals(addressIndex)) {
```

Comment:
I think it makes sense to add indexes for those fields for which we most often make selections. I think it would be better to do it in #312 for all tables